### PR TITLE
Enhanced - Handle the error status returned by platform APIs

### PR DESCRIPTION
**Pull request by stephenxs**
In case there is an error returned by platform API
- Translate the error bitmap to error description for generic errors
- Fetch the error description from sfp_error dict or via calling platform API get_error_description for vendor specific errors
- Adjust unit test cases

Signed-off-by: Stephen Sun <stephens@nvidia.com>

### CODE REVIEW

File: sonic-xcvrd/xcvrd/xcvrd_utilities/sfp_status_helper.py [hunk 1]
```diff
@@ -1,35 +1,38 @@
...
 def detect_port_in_error_status(logical_port_name, status_tbl):
     rec, fvp = status_tbl.get(logical_port_name)
     if rec:
         status_dict = dict(fvp)
-        if 'error' in status_dict:
-            for error in error_str_block_eeprom_reading:
-                if error in status_dict['error']:
-                    return True
+        error = status_dict.get('error')
+        if SfpBase.SFP_ERROR_DESCRIPTION_BLOCKING in error:
```
Comment (Junchao-Mellanox) [inline sonic-xcvrd/xcvrd/xcvrd_utilities/sfp_status_helper.py]: return SfpBase.SFP_ERROR_DESCRIPTION_BLOCKING in error

File: sonic-xcvrd/xcvrd/xcvrd.py [hunk 2]
```diff
@@ -759,22 +769,9 @@ def waiting_time_compensation_with_sleep(time_start, time_to_wait):
...
-        for error_code, error_msg in sfp_status_helper.SFP_STATUS_ERR_DICT.items():
-            if error_code & int_status:
-                error_list.append(error_msg)
-        if error_list:
-            fvs = swsscommon.FieldValuePairs([('status', str(int_status & 1)), ('error', '|'.join(error_list))])
-            status_tbl.set(logical_port_name, fvs)
-        else:
-            # SFP return unkown event, just ignore for now.
-            helper_logger.log_warning("Got unknown event {}, ignored".format(status))
+def update_port_transceiver_status_table(logical_port_name, status_tbl, status, error_descriptions='N/A'):
```
Comment (Junchao-Mellanox) [inline sonic-xcvrd/xcvrd/xcvrd.py:772]: It seems error_descriptions will never be 'N/A'

Comment (stephenxs) [reply sonic-xcvrd/xcvrd/xcvrd.py:772]: In a normal [insertion](https://github.com/Junchao-Mellanox/sonic-platform-daemons/pull/26/files/5a5aac4bbb6b7baba934365dcfc080b218f2b749#diff-fc70c43977a1ce57f3a6e9d7d09cb7ea5204f52b3adf3162936205e5680efb2fR1066)/[removal](https://github.com/Junchao-Mellanox/sonic-platform-daemons/pull/26/files/5a5aac4bbb6b7baba934365dcfc080b218f2b749#diff-fc70c43977a1ce57f3a6e9d7d09cb7ea5204f52b3adf3162936205e5680efb2fR1081) flow, the `error_descriptions` won't be fed therefore being `N/A`.

File: sonic-xcvrd/xcvrd/xcvrd_utilities/sfp_status_helper.py (hunk 1 above)
Comment (stephenxs) [reply sonic-xcvrd/xcvrd/xcvrd_utilities/sfp_status_helper.py]: Fixed

**Review by Junchao-Mellanox — APPROVED** [review]

**Pull request merged**